components/status_dropdown/status_dropdown.jsx

```diff
@@ -105,6 +120,87 @@ export default class StatusDropdown extends React.PureComponent {
             </FormattedMessage>
         );
     }
+    handleClearStatus = (e) => {
+        e.stopPropagation();
+        e.preventDefault();
+        this.props.actions.unsetCustomStatus();
+    };
+
+    onToggle = (open) => this.props.actions.setStatusDropdown(open);
+
+    renderCustomStatus = () => {
+        if (!this.props.isCustomStatusEnabled) {
+            return null;
+        }
+        const customStatus = this.props.customStatus;
+        const isStatusSet = customStatus && (customStatus.text || customStatus.emoji);
+        const customStatusText = isStatusSet ? customStatus.text : localizeMessage('status_dropdown.set_custom', 'Set a Custom Status');
+        const customStatusEmoji = isStatusSet ?
+            (
+                <span className='d-flex'>
+                    <CustomStatusEmoji
+                        showTooltip={false}
+                        emojiStyle={{marginLeft: 0}}
+                    />
+                </span>
+            ) : (
+                <EmojiIcon className={'custom-status-emoji'}/>
+            );
+
+        const clearButton = isStatusSet &&
+            (
+                <span
+                    className='custom_status__clear'
+                >
```

Comment:
I feel like this span isn't needed and would prefer not to have excess markup if not needed. Can we move the class to the child button and remove the span entirely?

Reply:
@deanwhillier Removing the span is not an issue, but the visual hover effect you mentioned is somewhat tricky.

Let me try to explain why.

Here is the status dropdown.
![image](https://user-images.githubusercontent.com/35728906/107972228-8bdc7780-6fd9-11eb-9811-73bc930d79f5.png)

The generated HTML looks something like this:

```html
<li>
    <button>status</button>
    <button>clear</button>
</li>
<li>
    <button>online</button>
</li>
<li>
    <button>away</button>
</li>
...
```


In the rest of the statuses (`online`, `away`), the hover effect is controlled by a CSS class on the button element.

Now, for the custom status as well, the hover state is controlled by the `status` button.

Since the `clear` button is added on top of the `status` button by using `position: absolute`, we can't detect the hover state on the `status` button while hovering over the `clear` button.

The alternative is to control the hover state using the `<li>` tag, which is a big change since we have to change and test the other status buttons as well.

Reply:
Yup, I totally get it! We'll be rebuilding the menu components in the near future so can revisit this later. Thanks for working through all the requested changes!